### PR TITLE
V2.23.0

### PR DESCRIPTION
**Pull request by NyakudyaA**
Upgrade to v2.23.0

### CODE REVIEW

File: build_data/web.xml [hunk 1]
```diff
@@ -596,36 +596,25 @@
...
     <filter>
-      <filter-name>CorsFilter</filter-name>
-      <filter-class>org.apache.catalina.filters.CorsFilter</filter-class>
-      <init-param>
-        <param-name>cors.allowed.origins</param-name>
-        <param-value>*</param-value>
-      </init-param>
-      <init-param>
-        <param-name>cors.allowed.methods</param-name>
-        <param-value>GET,POST,HEAD,OPTIONS,PUT</param-value>
```
Comment (tlvu) [inline build_data/web.xml:607]: Is there a reason why this is rollback from https://github.com/kartoza/docker-geoserver/pull/517?  It brokes some workflow?

Comment (NyakudyaA) [reply build_data/web.xml:607]: This breaks the image, tried to build it locally and it was erroring with those changes.

Comment (NyakudyaA) [reply build_data/web.xml:607]: @tlvu I get the following error when running with that XML
```
12-Apr-2023 18:40:11.018 SEVERE [main] org.apache.catalina.core.StandardContext.startInternal One or more Filters failed to start. Full details will be found in the appropriate container log file
12-Apr-2023 18:40:11.019 SEVERE [main] org.apache.catalina.core.StandardContext.startInternal Context [/geoserver] startup failed due to previous errors
12 Apr 18:40:11 CONFIG [gwc.layer] - GeoServer TileLayer store base directory is: gwc-layers
12 Apr 18:40:11 CONFIG [gwc.layer] - Loading tile layers from gwc-layers
12 Apr 18:40:11 CONFIG [gwc.layer] - Loaded 0 tile layers in 658.0 μs
12 Apr 18:40:11 INFO   [jdbc.JDBCQuotaStore] - Closing up the JDBC quota store
```

Comment (NyakudyaA) [reply build_data/web.xml:607]: and the tests do not pass with those changes

Comment (tlvu) [reply build_data/web.xml:607]: Wow I am surprised.  The param `cors.allowed.methods` was enabled for 2.19.0, for us, without any issues.

However we did not have `cors.support.credentials` and `cors.preflight.maxage` so they looked more suspect.

The rest of the changes, you simply changed the indentation so should not matter.

Comment (NyakudyaA) [reply build_data/web.xml:607]: I will try another build without cors.preflight.maxage and see if it works. By the way web.xml is completely customizable as a run time config so this doesn't break the image

**Pull request merged**